docs/compilers/terrapin.md

```diff
@@ -0,0 +1,125 @@
+.NET and Terrapin Process
+===
+The Terrapin system is concerned with validating the provenance of NuPkgs in the .NET ecosystem. Specifically it needs to verify that DLLs in a NuPkg can be rebuilt using publicly available artifacts. 
+
+This document describes the workflow that will be used by Terrapin for validating the provenance of .NET binaries, responsibilities of each team, as well as the tools the .NET team will deliver to enable the service to be successful.
+
+## .NET Tools
+The .NET Team will deliver a set of tools to the Terrapin team to provide the necessary validation. The idea in providing a series of tools here vs. a single all in one tool is to help better separate out the different responsibilities involved here and letting the appropriate team handle them:
+
+1. Providing the list of artifacts necessary for validation to succeed (owned by .NET)
+1. Acquiring the artifacts necessary for validation (owned by Terrapin)
+1. Performing validation and making a determination of provenance (owned by .NET)
+
+It's possible in the future we will end up providing a "all in one" experience tool but for the short term we'll be focusing on keeping these responsibilities and tools separate. 
+
+### dotnet-roslyn-manifest-generator
```

Comment:
Not clear from the name what kind of 'manifest' is this generating. Maybe something like `dotnet-roslyn-build-dependencies`? 

Does it need to be Roslyn specific? I think we could have a common format of the PDB records among C#, VB and F# (an d potentially other langs). Some entries in the records are universal across .NET: metadata references, runtime version, compiler version, language name. The rest are language specific but these wouldn't be needed by this tool.

So, I think we can have `dotnet-build-dependencies` tool that works with all managed binaries using a common set of entries that express dependencies.

Reply:
> Does it need to be Roslyn specific?

Long term I think that terrapin will end up encompassing all of the tools in our build that contribute to how our DLL files are written. So there are some obvious examples that go past Roslyn: single file writer, F#, etc ... Hence I do think long term we will need a manifest file that encompasses the artifacts needed be the totality of these tools. 

At the same time I tried to write this section to be more general but had trouble doing so as we don't know the exact requirements of the tools. Given that the next six months is largely Terrapin focused I kept this doc fairly Roslyn specific to keep it concrete. 

I should add a section at the end though clarifying that the intent here is for this to envolve though. 

> So, I think we can have dotnet-build-dependencies tool that works with all managed binaries using a common set of entries that express dependencies.

As the end of the doc states, I'm terrible at naming. I'm fine with whatever name others agree on. @clairernovotny do u have any opinions here?